app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -1735,4 +1735,46 @@ class PatientRepositoryAndroidTest {
     assertThat(latestBpPassport).isEqualTo(currentBpPassport)
   }
 
+  @Test
+  fun when_updating_patient_recordedAt_then_it_should_compare_and_update_the_date() {
+
+    fun createPatientProfile(patientUuid: UUID, recordedAt: Instant): PatientProfile {
+      return testData.patientProfile(patientUuid = patientUuid)
+          .run {
+            copy(patient = patient.copy(
+                recordedAt = recordedAt,
+                updatedAt = Instant.now(testClock),
+                syncStatus = DONE)
+            )
+          }
+    }
+
+    val patientUuid1 = UUID.randomUUID()
+    val recordedAtForPatient1 = Instant.now(testClock)
+    val patientProfile1 = createPatientProfile(patientUuid1, recordedAtForPatient1)
+    testClock.advanceBy(Duration.ofMinutes(1))
+    val bpRecordedAtForPatient1 = Instant.now(testClock)
+
+    val patientUuid2 = UUID.randomUUID()
+    val bpRecordedAtForPatient2 = Instant.now(testClock)
+    testClock.advanceBy(Duration.ofMinutes(1))
+    val recordedAtForPatient2 = Instant.now(testClock)
+    val patientProfile2 = createPatientProfile(patientUuid2, recordedAtForPatient2)
+
+    patientRepository.save(listOf(patientProfile1, patientProfile2)).blockingAwait()
+
+    patientRepository.updatePatientRecordedAt(patientUuid1, bpRecordedAtForPatient1).blockingGet()
```

Comment:
Using `blockingGet()` makes me think that this method is returning a `Single<>` when it's actually a `Completable` which will return a `Throwable|null`. Should we change this to `blockingAwait()` instead?